cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -299,6 +248,32 @@ func startControllers(c *cloudcontrollerconfig.CompletedConfig, stop <-chan stru
 	select {}
 }
 
+// IsControllerEnabled check if a specified controller enabled or not.
+func IsControllerEnabled(name string, disabledByDefaultControllers sets.String, controllers ...string) bool {
+	hasStar := false
+	for _, ctrl := range controllers {
+		if ctrl == name {
+			return true
+		}
+		if ctrl == "-"+name {
+			return false
+		}
+		if ctrl == "*" {
+			hasStar = true
+		}
+	}
+	// if we get here, there was no explicit choice
+	if !hasStar {
+		// nothing on by default
+		return false
+	}
+	if disabledByDefaultControllers.Has(name) {
```

Comment:
return !disabledByDefaultControllers.Has(name)

Reply:
seems no need:  if `disabledByDefaultControllers.Has(name)` is `true`, will return `false`, else will return `true` in line276